### PR TITLE
Ticket1124 device screens list

### PR DESCRIPTION
**Pull request by John-Holt-Tessella**
###Description of work

Screen to show device screen list from PV. When devices are clicked an OPI is opened.

###To test

For ticket https://github.com/ISISComputingGroup/IBEX/issues/1124
Also see: https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/54
System tests: https://github.com/ISISComputingGroup/ibex_system_tests/pull/19
###Acceptance criteria

1. Device are displayed in the list which appear in the device screen xml.
    1. XML can be edited by adding it to text box (white area).
    1, Example xml below
1. Device screen updates when PVs update
1. Device screen  changes with different instruments
1. Device screen can be sorted by clicking on titles
1. Device screen has appropriate icons for various devices (key is opi resource key)
1. When item is clicked OPI is shown
    1. OPI is correct opi
    1. Macros are set
    1. Tab has the correct title
1. Multiple devices can be shown
1. When item is clicked for a second time OPI is brought to the front
1. OPI can be closed
1. When instrument is switched open OPIs are closed.
1. Check synoptic OPIs still work.

Example XML
```
<?xml version="1.0" ?>
<devices xmlns="http://epics.isis.rl.ac.uk/schema/screens/1.0/">
	<device>
		<name>My Kepco</name>
		<key>Kepco</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>IOC_NUM</key>
				<value>01</value>
			</property>
			<property>
				<key>Property2</key>
				<value>Value2</value>
			</property>
		</properties>
	</device>
	<device>
		<name>Attenuator</name>
		<key>Attenuator</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>Q</key>
				<value>ATTEN</value>
			</property>
		</properties>
	</device>
	<device>
		<name>Attenuator B</name>
		<key>Attenuator</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>Q</key>
				<value>ATTEN</value>
			</property>
		</properties>
	</device>	
	<device>
		<name>Error IOC</name>
		<key>MyDeviceKey</key>
		<type>COMPONENT</type>
		<properties>
			<property>
				<key>Property1</key>
				<value>Value1</value>
			</property>
			<property>
				<key>Property2</key>
				<value>Value2</value>
			</property>
		</properties>
	</device>
</devices>
```

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [x] Are there [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has developer documentation been updated if required?

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.devicescreens.tests/src/uk/ac/stfc/isis/ibex/devicescreens/desc/tests/DeviceDescriptionTest.java [hunk 1]
```diff
@@ -115,4 +116,42 @@ public void GIVE_new_device_description_THEN_properties_can_be_added() {
...
+        deviceDescription.addProperty(property1);
+        deviceDescription.addProperty(property2);
+
+        // Act
+        OpiTarget result = deviceDescription.getOPITarget();
+
+        // Assert
+        assertEquals("Number of propeties", result.properties().size(), 2);
+        assertEquals("First key value", result.properties().get(key1), value1);
+        assertEquals("First key value", result.properties().get(key2), value2);
```
Comment (AdrianPotter) [inline base/uk.ac.stfc.isis.ibex.devicescreens.tests/src/uk/ac/stfc/isis/ibex/devicescreens/desc/tests/DeviceDescriptionTest.java]: Surely this is second key value?

File: base/uk.ac.stfc.isis.ibex.synoptic/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -14,7 +14,9 @@ Require-Bundle: org.eclipse.core.runtime,
  uk.ac.stfc.isis.ibex.instrument,
  uk.ac.stfc.isis.ibex.json;bundle-version="1.0.0",
  uk.ac.stfc.isis.ibex.configserver;bundle-version="1.0.0",
- uk.ac.stfc.isis.ibex.epics.switching
+ uk.ac.stfc.isis.ibex.epics.switching,
+ uk.ac.stfc.isis.ibex.devicescreens,
+ uk.ac.stfc.isis.ibex.targets;bundle-version="1.0.0"
```
Comment (AdrianPotter) [inline base/uk.ac.stfc.isis.ibex.synoptic/META-INF/MANIFEST.MF]: Why does targets get a specific bundle version?

File: base/uk.ac.stfc.isis.ibex.ui.synoptic/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -20,11 +20,13 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.ui.ide,
  uk.ac.stfc.isis.ibex.ui.dae;bundle-version="1.0.0",
  uk.ac.stfc.isis.ibex.instrument;bundle-version="1.0.0",
- uk.ac.stfc.isis.ibex.configserver;bundle-version="1.0.0"
+ uk.ac.stfc.isis.ibex.configserver;bundle-version="1.0.0",
+ uk.ac.stfc.isis.ibex.devicescreens,
+ uk.ac.stfc.isis.ibex.ui.devicescreens;bundle-version="1.0.0"
```
Comment (AdrianPotter) [inline base/uk.ac.stfc.isis.ibex.ui.synoptic/META-INF/MANIFEST.MF]: As before: why do some plugins get specified version numbers and not others?

File: base/uk.ac.stfc.isis.ibex.synoptic/META-INF/MANIFEST.MF (hunk 2 above)
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.synoptic/META-INF/MANIFEST.MF]: This is a minimum version but I think the difference is whether it is right click add and add in manifest.

**Pull request merged**